photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+            to collect the data to do the fitting
+        """
+
+        self.find = find
+        self.group = group
+        self.bkg = bkg
+        self.psf = psf
+        self.fitter = fitter
+        self.niters = niters
+        self.fitshape = fitshape
+
+        @property
+        def niters(self):
+            return self._niters
+        @niters.setter
+        def niters(self, niters):
+            if isinstance(niters, int) and niters > 0:
+                self._niters = niters
+            else:
+                raise ValueError('niters is not defined properly, '
+                                 'received niters = {}'.format(niters))
+        @property
+        def fitshape(self):
+            return self._fitshape
+        @fitshape.setter
+        def fitshape(self, fitshape):
+            fitshape = np.asarray(fitshape)
+            if len(fitshape) == 2 and np.all(fitshape) > 0:
+                self._fitshape = fitshape
+            else:
+                raise ValueError('fitshape is not defined properly, '
+                                 'received fitshape = {}'.format(fitshape))
+
+    def __call__(self, image):
+        """
+        Parameters
+        ----------
+        image : array-like, ImageHDU, HDUList
+            image to perform photometry
+        
+        Returns
+        -------
+        outtab : astropy.table.Table
+            Table with the photometry results, i.e., centroids and flux
+            estimations.
+        residual_image : array-like, ImageHDU, HDUList
+            Residual image calculated by subtracting the fitted sources
+            and the original image.
+        """
+
+        return self.do_photometry(image)
+
+    def _nstar(self, image, star_groups):
+        """
+        Fit, as appropriate, a compound or single model to a given `groups` of
+        stars. Groups are fitted sequentially from the smallest to the biggest. In
+        each iteration, `image` is subtracted by the previous fitted group. 
+        
+        Parameters
+        ----------
+        image : numpy.ndarray
+            Background-subtracted image.
+        star_groups : list of `~astropy.table.Table`
+            Each `~astropy.table.Table` in this list corresponds to a group of
+            mutually overlapping starts.
+
+        Return
+        ------
+        result_tab : `~astropy.table.Table`
+            Astropy table that contains the results of the photometry.
+        image : numpy.ndarray
+            Residual image.
+        """
+
+        result_tab = Table([[], [], [], [], []],
+                           names=('id', 'group_id', 'x_fit', 'y_fit',
+                                  'flux_fit'),
+                           dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
+
+        star_groups = star_groups.group_by('group_id')
+        
+        groups_order = []
+        for g in star_groups.groups:
+            groups_order.append(len(g))
```

Comment:
Does this really matter?
Stetson did smallest to biggest, but I thought the groups are independent, so the order should (?) not matter. If you just use rthe order of "group_id' you a) save a few lines and b) allow the user a way to order groups in any way they like if necessary (by relabelling the colum group_id).
I assume that Stetson did smallest-to-largest because larger groups are more likely to crash the fit. That's still true, but since we assemble the results table in memory (while Stetson writes every line to disk when it's fit) we implicitly require that the code runs all the way through anyway.
Instead, I suggest something like the following sentence in the docstring:

```
Notes
-------
If there are problems with fitting large groups, change the parameters of the grouping algorithm to reduce the number of sources in each group or input a ``star_groups`` table that only includes the groups that are relevant (e.g. manually remove all entries that coincide with artifacts).
```